Repository/Enrollments/EnrollmentRepository.cs

```diff
@@ -16,86 +17,118 @@ public EnrollmentRepository(DbModels.DbModels context)
             this.context = context;
         }
 
-        public EnrollmentDTO Add(EnrollmentDTO enrollment)
+        public EnrollmentDTO AddEnrollment(EnrollmentDTO enrollment)
         {
-            context.Enrollments.Add(new DbModels.Enrollment
+            try
             {
-                ID = enrollment.ID,
-                CourseID = enrollment.CourseID,
-                UserID = enrollment.UserID,
-                Grade = enrollment.Grade,
-                
-            });
-            context.SaveChanges();
-            return enrollment;
-        }
-
-        public EnrollmentDTO Delete(int ID)
-        {
-            Enrollment enrollment = context.Enrollments.Find(ID);
-            if (enrollment != null)
-            {
-                var deletedEnrollment = new EnrollmentDTO
+                context.Enrollments.Add(new DbModels.Enrollment
                 {
                     ID = enrollment.ID,
                     CourseID = enrollment.CourseID,
                     UserID = enrollment.UserID,
                     Grade = enrollment.Grade,
-                };
-                context.Enrollments.Remove(enrollment);
+
+                });
                 context.SaveChanges();
-                return deletedEnrollment;
+                return enrollment;
+            }
+            catch (Exception ex)
+            {
+                Trace.WriteLine(ex.Message);
+            }
+            return null;
+        }
+
+        public EnrollmentDTO DeleteEnrollment(int ID)
+        {
+            try
+            {
+                Enrollment enrollment = context.Enrollments.Find(ID);
+                if (enrollment != null)
+                {
+                    var deletedEnrollment = new EnrollmentDTO
+                    {
+                        ID = enrollment.ID,
+                        CourseID = enrollment.CourseID,
+                        UserID = enrollment.UserID,
+                        Grade = enrollment.Grade,
+                    };
+                    context.Enrollments.Remove(enrollment);
+                    context.SaveChanges();
+                    return deletedEnrollment;
+                }
+                return null;
```

Comment:
There is no need for this `return null;`. You already wrote the same logic below the `catch` statement, meaning, if enrollment doesn't exist, it will continue execution and line 66 will be executed